core/Azure.Mcp.Core/src/Areas/Server/Commands/ToolLoading/NamespaceToolLoader.cs

```diff
@@ -288,6 +289,67 @@ private async Task<CallToolResult> InvokeChildToolAsync(
                 return await InvokeToolLearn(request, intent, namespaceName, cancellationToken);
             }
 
+            // Check if this tool requires elicitation for sensitive data
+            var metadata = cmd.Metadata;
```

Comment:
This change looks the same as the logic in `CommandFactoryToolLoader`. It would be nice to share the logic.